src/mod/endpoints/mod_sofia/sofia_reg.c

```diff
@@ -575,7 +576,9 @@ void sofia_reg_check_gateway(sofia_profile_t *profile, time_t now)
 			{
 				int sec;
 
-				if (gateway_ptr->fail_908_retry_seconds && gateway_ptr->failure_status == 908) {
+				if(gateway_ptr->deleted) {
```

Comment:
I don't get why changing the sec to 0 when the gateway is already mark as deleted? Changing it to 0 means it will switch out immediately to new state `REG_STATE_UNREGED`. Also I don't think that a gateway that is mark as deleted can still be in the gateway list since `sofia_reg_truly_del_gateway` should have clean it out before we can start iterating to the gateway list